app/uk/gov/hmrc/helptosave/controllers/HelpToSaveController.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosave.controllers
+
+import cats.instances.int._
+import cats.syntax.eq._
+import com.google.inject.Inject
+import play.api.libs.json.{JsError, JsSuccess}
+import play.api.mvc.{Action, AnyContent, Result}
+import uk.gov.hmrc.helptosave.config.AppConfig
+import uk.gov.hmrc.helptosave.connectors.{HelpToSaveProxyConnector, ITMPEnrolmentConnector}
+import uk.gov.hmrc.helptosave.models.{ErrorResponse, NSIUserInfo}
+import uk.gov.hmrc.helptosave.repo.EnrolmentStore
+import uk.gov.hmrc.helptosave.services.{EligibilityCheckService, UserCapService}
+import uk.gov.hmrc.helptosave.util.JsErrorOps._
+import uk.gov.hmrc.helptosave.util.Logging._
+import uk.gov.hmrc.helptosave.util.{LogMessageTransformer, WithMdcExecutionContext, toFuture}
+import uk.gov.hmrc.play.bootstrap.controller.BaseController
+
+import scala.util.{Failure, Success}
+
+class HelpToSaveController @Inject() (val enrolmentStore:          EnrolmentStore,
+                                      val itmpConnector:           ITMPEnrolmentConnector,
+                                      proxyConnector:              HelpToSaveProxyConnector,
+                                      userCapService:              UserCapService,
+                                      val eligibilityCheckService: EligibilityCheckService)(
+    implicit
+    transformer: LogMessageTransformer, appConfig: AppConfig)
+  extends BaseController with EligibilityBase with EnrolmentBehaviour with WithMdcExecutionContext {
+
+  def createAccount(): Action[AnyContent] = Action.async {
```

Comment:
For these endpoints I think there is a minimum amount of authentication we can do which is to require the AuthProvider to be either `GovernmentGateway` or `PrivilegedApplication`.

Reply:
e.g.
```scala
authorised(AuthProviders(GovernmentGateway, PrivilegedApplication)){ ... }
```

Reply:
I thought about this, but how to check for the auth incase of stride requests ?

Reply:
nvm, i just realised stride auth uses `PrivilegedApplication` type too , so is the same for API quests too ?

Reply:
is it ok if i add auth to these endpoints in a upcoming PR  ?

Reply:
yeah, stride and privileged access through the API would come under `PrivilegedApplication` - let's add the auth to your next PR then